modules/siddhi-extensions/output-mappers/json-output-mapper/src/main/java/org/wso2/siddhi/extension/output/mapper/json/JsonSinkMapper.java

```diff
@@ -0,0 +1,361 @@
+     * @param payloadTemplateBuilder Unmapped payload for reference
+     */
+    @Override
+    public void init(StreamDefinition streamDefinition, OptionHolder optionHolder,
+                     TemplateBuilder payloadTemplateBuilder, ConfigReader mapperConfigReader) {
+        attributeNameArray = streamDefinition.getAttributeNameArray();
+        enclosingElement = optionHolder.validateAndGetStaticValue(ENCLOSING_ELEMENT_IDENTIFIER, null);
+        isJsonValidationEnabled = Boolean.parseBoolean(optionHolder
+                .validateAndGetStaticValue(JSON_VALIDATION_IDENTIFIER, "false"));
+    }
+
+    @Override
+    public void mapAndSend(Event[] events, OptionHolder optionHolder, TemplateBuilder payloadTemplateBuilder,
+                           SinkListener sinkListener, DynamicOptions dynamicOptions)
+            throws ConnectionUnavailableException {
+        StringBuilder sb = null;
+        if (payloadTemplateBuilder == null) {
+            String jsonString = constructJsonForDefaultMapping(events);
+            if (jsonString != null) {
+                sb = new StringBuilder();
+                sb.append(jsonString);
+            }
+        } else {
+            sb = new StringBuilder();
+            sb.append(constructJsonForCustomMapping(events, payloadTemplateBuilder));
+        }
+
+        if (sb != null) {
+            if (!isJsonValidationEnabled) {
+                sinkListener.publish(sb.toString(), dynamicOptions);
+            } else if (isValidJson(sb.toString())) {
+                sinkListener.publish(sb.toString(), dynamicOptions);
+            } else {
+                log.error("Invalid json string : " + sb.toString() + ". Hence dropping the message.");
+            }
+        }
+    }
+
+    @Override
+    public void mapAndSend(Event event, OptionHolder optionHolder, TemplateBuilder payloadTemplateBuilder,
+                           SinkListener sinkListener, DynamicOptions dynamicOptions)
+            throws ConnectionUnavailableException {
+        StringBuilder sb = null;
+        if (payloadTemplateBuilder == null) {
+            String jsonString = constructJsonForDefaultMapping(event);
+            if (jsonString != null) {
+                sb = new StringBuilder();
+                sb.append(jsonString);
+            }
+        } else {
+            sb = new StringBuilder();
+            sb.append(constructJsonForCustomMapping(event, payloadTemplateBuilder));
+        }
+
+        if (sb != null) {
+            if (!isJsonValidationEnabled) {
+                sinkListener.publish(sb.toString(), dynamicOptions);
+            } else if (isValidJson(sb.toString())) {
+                sinkListener.publish(sb.toString(), dynamicOptions);
+            } else {
+                log.error("Invalid json string : " + sb.toString() + ". Hence dropping the message.");
+            }
+        }
+    }
+
+    private String constructJsonForDefaultMapping(Object eventObj) {
```

Comment:
The same thing is already happening within this method. In the case of event arrays, for each event, constructSingleEventForDefaultMapping() method will be called. The reason to use a separate method to include these was to keep the parent method clean without complicated code segments.